README.md

```diff
@@ -160,39 +229,53 @@ git clone https://github.com/bao-project/bao-hypervisor $BAO_SRCS\
+
 ```sh
 mkdir -p $mkdir -p $BUILD_BAO_DIR/config
-cp -L $ROOT_DIR/configs/baremetal.c\
-    $BUILD_BAO_DIR/config/baremetal.c
+cp -L $ROOT_DIR/configs/baremetal.c $BUILD_BAO_DIR/config/baremetal.c
 ```
 
 #### 2.2.3. Compiling Bao Hypervisor
-With all set, it's time to bring your Bao Hypervisor to life. You now just need to compile it!
+We are all set! It's time to bring the Bao Hypervisor to life. Now, we just
+need to compile it!
+
 ```sh
 make -C $BAO_SRCS\
     PLATFORM=qemu-aarch64-virt\
     CONFIG_REPO=$ROOT_DIR/configs\
     CONFIG=baremetal\
     CONFIG_BUILTIN=y\
-    CPPFLAGS=-DBAO_WRKDIR_IMGS=$SETUP_BUILD
+    CPPFLAGS=-DBAO_WRKDIR_IMGS=$BUILD_GUESTS_DIR
 ```
 
-Upon completing these steps, you'll find a binary file in the BAO_SRCS directory, called bao.bin. Now, move the binary file to your build directory (BUILD_BAO_DIR):
+Upon completing these steps, you'll find a binary file in the `BAO_SRCS`
+directory called `bao.bin`. Now, let's move the binary file to your build
+directory:
 
 ```sh
 cp $BAO_SRCS/bin/qemu-aarch64-virt/baremetal/bao.bin $BUILD_BAO_DIR/bao.bin
 ```
 
 ## 3. Build Firmware - Powering Up Your Setup
 
-No journey is truly complete without firmware. It's the fuel that powers your virtual world. That's why we're here to guide you through acquiring the essential firmware tailored to your target platform (you can find the pointer to build the firmware to other platforms [here](https://github.com/bao-project/bao-demos#b5-build-firmware-and-deploy)).
+Firmware is essential to power your virtual world. That's why we're here to
+assist you in obtaining the necessary firmware tailored to your target platform
+(you can find the instructions to build the firmware for other platforms
+[here](https://github.com/bao-project/bao-demos#b5-build-firmware-and-deploy)).
 
-### 3.1 Welcome to the QEMU platform!
+### 3.1 Getting Started with the QEMU Platform
 
-Why bother with a hardware platform when you have QEMU? If you haven't got it yet, fret not. We're here to guide you through the process of building and installing it. In this guide, our focus will be on Aarch64 QEMU.
+QEMU provides a convenient alternative to a hardware platform. If you haven't
+installed it yet, don't worry. We're here to walk you through the process of
+building and installing it. In this guide, we'll target the Aarch64
+architecture.
 
-However, if you're already equipped with qemu-system-aarch64, or if compiling isn't your cup of tea and you'd rather install it directly using a package manager or another method, ensure that you're working with version 7.2.0 or higher. In that case, you can jump ahead to the next step.
+If you already have qemu-system-aarch64, or if you'd prefer to install it
+directly using a package manager or another method, just make sure you're
+working with version 7.2.0 or higher. In that case, feel free to move on to the
+[next step](#32-now-you-need-u-boot).
 
 To install QEMU, simply run the following commands:
 
```

Comment:
On the commands below, I believe this command is repeated:
``mkdir -p $ROOT_DIR/tools/bin``

Reply:
You're correct, fixed!